lib/lambda.js

```diff
@@ -100,19 +79,12 @@ module.exports = function elasticApmAwsLambda (agent) {
       })
 
       // Wrap context and callback to finish and send transaction
-      context = wrapContext(trans, payload, context)
+      wrapContext(trans, payload, context)
       if (typeof callback === 'function') {
         callback = wrapLambdaCallback(trans, payload, context, callback)
       }
 
-      const result = fn.call(this, payload, context, callback)
-
-      return result && typeof result.then === 'function'
-        ? result.then(
-          passFn(trans, payload, context),
-          failFn(trans, payload, context)
-        )
-        : result
+      return fn.call(this, payload, context, callback)
```

Comment:
How will the transaction be ended in case there's no `callback`?

Reply:
The context methods will either be called directly by the user, or internally via the promise execution.